nilearn/maskers/nifti_labels_masker.py

```diff
@@ -291,28 +367,14 @@ def _reporting(self):
             if "warning_message" in self._report_content:
                 self._report_content["warning_message"] = None
 
-            labels_image = image.load_img(labels_image, dtype="int32")
-            labels_image_data = image.get_data(labels_image)
-            labels_image_affine = labels_image.affine
-            # Number of regions excluding the background
-            number_of_regions = np.sum(
-                np.unique(labels_image_data) != self.background_label
-            )
-            # Basic safety check to ensure we have as many labels as we
-            # have regions (plus background).
-            if (
-                self.labels is not None
-                and len(self.labels) != number_of_regions + 1
-            ):
-                raise ValueError(
-                    "Mismatch between the number of provided labels "
-                    f"({len(self.labels)}) and the number of regions in "
-                    f"provided label image ({number_of_regions + 1})."
-                )
```

Comment:
extracted this in separate methods so they can be used for checking the labels in the constructor.